packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -630,69 +637,111 @@ class SrvImplDart implements Srv<SocketConnector> {
     return socketConnector;
   }
 
-  Future _handleMultiConnectRequest(
+  Future<void> _handleMultiConnectRequest(
+    SocketConnector sc,
+    InternetAddress relayAddress,
+    DataTransformer? encrypter,
+    DataTransformer? decrypter,
+  ) async {
+    logger.info('_runDaemonSideMulti'
+        ' Control socket received connect request - '
+        ' creating new socketToSocket connection');
+
+    InternetAddress localAddress = await resolveRequestedLocalHost();
+
+    // First, connect to the relay.
+    // If it fails, return, as we can't do anything more
+    late Socket sideBSocket;
+    late Side sideB;
+    logger.info(
+        'socket_connector: Connecting side B (relay - $relayAddress:$streamingPort)');
+    try {
+      sideBSocket = await Socket.connect(relayAddress, streamingPort);
+      sideB = Side(sideBSocket, false, transformer: decrypter);
+      unawaited(sideBSocket.done
+          .then((v) => logger.info('relay socket done'))
+          .catchError(
+              (err) => logger.warning('relay socket done with error $err')));
+      if (rvdAuthString != null) {
+        logger.info('_runDaemonSideMulti authenticating'
+            ' new socket connection to relay');
+        sideBSocket.writeln(rvdAuthString);
+        await sideBSocket.flush();
+      }
+    } catch (e) {
+      logger.shout(
+          'Failed to connect to relay ($relayAddress:$streamingPort) with error : $e');
+      return;
+    }
+
+    // Now, connect to the local host:port
+    // If it fails, we need to close the socket connection to the relay
+    late Socket sideASocket;
+    late Side sideA;
+    logger.info(
+        'socket_connector: Connecting side A (local - $localAddress:$localPort)');
+
+    try {
+      sideASocket = await Socket.connect(localAddress, localPort);
+      sideA = Side(sideASocket, true, transformer: encrypter);
+    } catch (e) {
+      logger.shout(
+          'Failed to connect locally ($localAddress:$localPort) with error : $e');
+      logger.shout('Closing sideB (relay) socket connection');
+      sideBSocket.destroy();
+
+      return;
+    }
+
+    // Finally, let's have the socket connector join the sides together
+    unawaited(sc.handleSingleConnection(sideB).catchError((err) {
+      logger.severe(
+          'ERROR $err from handleSingleConnection on sideB (relay - $relayAddress:$streamingPort)');
+    }));
+    unawaited(sc.handleSingleConnection(sideA).catchError((err) {
+      logger.severe(
+          'ERROR $err from handleSingleConnection on sideA (local - $localAddress:$localPort)');
+    }));
```

Comment:
Once we've added both sides, socket connector will join them up so traffic starts to flow from local socket to relay socket (and within the relay, from this side's socket to the client side's socket, and on the client side from its relay socket to its local socket.)